packages/core-modules/test/contracts/modules/FeatureFlagModule.test.js

```diff
@@ -0,0 +1,137 @@
+  });
+
+  it('does not allow non-owners to set feature flag addresses', async () => {
+    await assertRevert(
+      FeatureFlagModule.connect(user).addToFeatureFlag(FEATURE_FLAG_NAME, permissionedUser.address),
+      'Unauthorized'
+    );
+  });
+
+  let addAddressTx;
+  before('setup permissioned user for feature flag', async () => {
+    addAddressTx = await FeatureFlagModule.connect(owner).addToFeatureFlag(
+      FEATURE_FLAG_NAME,
+      permissionedUser.address
+    );
+  });
+
+  it('emits event for adding address', async () => {
+    await assertEvent(
+      addAddressTx,
+      `FeatureFlagAddressAdded("${FEATURE_FLAG_NAME}", "${permissionedUser.address}")`,
+      FeatureFlagModule
+    );
+  });
+
+  describe('call function behind feature flag', async () => {
+    it('reverts when user does not have permission', async () => {
+      await assertRevert(
+        SampleFeatureFlagModule.connect(user).setFeatureFlaggedValue(15),
+        'Unauthorized'
+      );
+    });
+
+    it('allows permissioned user to call function', async () => {
+      await SampleFeatureFlagModule.connect(permissionedUser).setFeatureFlaggedValue(15);
+      assertBn.equal(await SampleFeatureFlagModule.getFeatureFlaggedValue(), 15);
+    });
+  });
+
+  it('does not allow non-owners to remove feature flag addresses', async () => {
+    await assertRevert(
+      FeatureFlagModule.connect(user).removeFromFeatureFlag(
+        FEATURE_FLAG_NAME,
+        permissionedUser.address
+      ),
+      'Unauthorized'
+    );
+
+    await assertRevert(
+      FeatureFlagModule.connect(permissionedUser).removeFromFeatureFlag(
+        FEATURE_FLAG_NAME,
+        permissionedUser.address
+      ),
+      'Unauthorized'
+    );
+  });
+
+  describe('remove permissioned user from feature flag', async () => {
+    let removeAddressTx;
+
+    before('remove user', async () => {
+      removeAddressTx = await FeatureFlagModule.connect(owner).removeFromFeatureFlag(
+        FEATURE_FLAG_NAME,
+        permissionedUser.address
+      );
+    });
+
+    it('emits event', async () => {
+      await assertEvent(
+        removeAddressTx,
+        `FeatureFlagAddressRemoved("${FEATURE_FLAG_NAME}", "${permissionedUser.address}")`,
+        FeatureFlagModule
+      );
+    });
+
+    it('does not allow removed permissionedUser to set value', async () => {
+      await assertRevert(
+        SampleFeatureFlagModule.connect(permissionedUser).setFeatureFlaggedValue(25),
+        'Unauthorized'
+      );
+    });
+  });
+
+  describe('disable feature flag for all', async () => {
+    before('disable feature flag', async () => {
+      await FeatureFlagModule.connect(owner).setFeatureFlag(FEATURE_FLAG_NAME, false);
+    });
+
+    it('reverts when feature flag is disabled', async () => {
+      await assertRevert(
+        SampleFeatureFlagModule.connect(permissionedUser).setFeatureFlaggedValue(25),
+        'FeatureUnavailable'
```

Comment:
you will likely need to put the contract which has the ABI containin the `FeatureUnavailable` flag in order for parsing... or have you not found that necessary?